lib/trust_level.rb

```diff
@@ -31,32 +31,23 @@ def self.name(level)
 
   def self.calculate(user, use_previous_trust_level: false)
     # First, use the manual locked level
-    if user.manual_locked_trust_level.present?
-      return user.manual_locked_trust_level
-    end
+    return user.manual_locked_trust_level if user.manual_locked_trust_level.present?
 
     # Then consider the group locked level (or the previous trust level)
-    granted_trust_level = user.group_granted_trust_level
+    granted_trust_level = user.group_granted_trust_level || 0
+    previous_trust_level = use_previous_trust_level && find_previous_trust_level(user) || 0
```

Comment:
I think the ternary reads much easier/better than the double conditional

```suggestion
    previous_trust_level = use_previous_trust_level ? find_previous_trust_level(user) : 0
```